src/netbase.cpp

```diff
@@ -756,7 +756,7 @@ std::string NetworkErrorString(int err)
 std::string NetworkErrorString(int err)
 {
     char buf[256];
-    const char *s = buf;
+    const char *s = nullptr;
     buf[0] = 0;
 /* Too bad there are two incompatible implementations of the
  * thread-safe strerror. */
```

Comment:
If the POSIX path is taken, s is still set as nullptr so the strprintf() show the error.
I think you need "s = buf" in the POSIX path

Reply:
Indeed! I changed it so that it is properly set in the non-POSIX path but explicitly only in that code path.